table/sst_file_writer.cc

```diff
@@ -72,19 +75,19 @@ SstFileWriter::~SstFileWriter() {
     // abandon the builder.
     rep_->builder->Abandon();
   }
-
-  delete rep_;
 }
 
 Status SstFileWriter::Open(const std::string& file_path) {
```

Comment:
Actually I thought about this again and the reason I made it a constructor argument is that it is conceptually similar to the other options passed there, like EnvOptions and invalidate_page_cache, are you fine with landing it in this current form?